---
layout: mdpage
---

## STAR Job

STAR is a tool used to map sequence reads against reference genomes. Job files which include ```job: star``` signal Biowizard to use STAR to map reads.

### Inputs
    
    inputs:
        genome: "s3://my-s3-bucket/genome-indexes/mygenomeindex.tar.gz"
        fastq: "s3://my-s3-bucket/test_100reads.fastq.gz"

    inputs:
        genome: "s3://my-s3-bucket/genome-indexes/mygenomeindex.tar.gz"
        fastq: 
            - "s3://my-s3-bucket/test_100reads.fastq.gz"
            - "s3://my-s3-bucket/test_101reads.fastq.gz"
            - "s3://my-s3-bucket/test_102reads.fastq.gz"


#### Valid Input Parameters

* ```genome``` *Required* Genome directory, genereated by STAR --generateGenome. Provided in .tar archive.
* ```fastq``` *Required* One or more files containing sequence reads in fastq format.

* ```runMode``` genomeGenerate
* ```sjdbGTFfile``` Annotations file
* ```sjdbOverhang``` e.g. ReadLength-1

### Outputs

Biowizard accepts several output options for sequence mapping jobs using STAR.

    outputs:
        sorted_bam: "s3://my-s3-bucket/sorted.bam"
        sorted_bam_index: "s3://my-s3-bucket/sorted.bam.bai"
        logs: "s3://my-s3-bucket/logs"

#### Valid Output Parameters

* ```logs``` *Optional* Log files generated by this job will be stored to this location
* ```sorted_bam``` *Optional* Mapped reads, converted to BAM format and sorted
* ```sorted_bam_bai``` *Optional* .bai index of sorted BAM file.;Hello dere


